### DISCUSSION TITLE
How to build JS in a separate process without rails/webpack fighting?

### DISCUSSION THREAD

**Discussion by BenSturmfels** [houdiniproject/houdini#1053] [Q&A] [answered]
The `docs/getting_started.md` advises building JS in a separate `webpack --watch` process during development. When I do this though, `rails server` continues to also compile JS on the next request after changes to the JS code. How do you tell Rails not to also build the JS?

**Comment by wwahammy** [comment, marked as answer]
Generally, you wouldn't want to tell Rails not to build the Javascript in development. You can run `bin/webpack` to prebuild it though and, if none of the source files have changed, it probably would just pass the built assets through to the end-user. 

The advice on using `webpack --watch` is out of date.

**Reply by BenSturmfels** [reply]
I've now updated the `getting_started.md` docs to mention `bin/webpack-dev-server` in #1051. I can confirm that there doesn't seem to be any double-building going when using this approach.
